php/libraries/CouchDB.class.inc

```diff
@@ -70,7 +70,14 @@ class SocketWrapper
      */
     function open()
     {
-        $this->socket = fsockopen($this->host, $this->port);
+        if (strpos($this->host, "%") !== false) { // %COUCH_HOSTNAME%
+            $this->socket = false;
+        }
+        if (!is_numeric($this->port)) { // %COUCH_PORT%
+            $this->socket = false;
+        } else {
+            $this->socket = fsockopen($this->host, $this->port);
+        }
```

Comment:
What are you trying to accomplish with this? Opening a connection to a CouchDB server that isn't configured is meaningless and this just pushes the error further downstream when someone tries to do the next operation on the socket.

Reply:
Just a console output cleanup. The throw new LorisException does the job by showing the error:
> The following errors occured while attempting to display this page:
>
> Could not connect to CouchDB server at %COUCH_HOSTNAME%:%COUCH_PORT%

A default untouched configuration (where the CouchDB hasn't been configured) will display the console warning: fsockopen() expects parameter 2 to be integer, string given on line 73. As well with the LorisException.

Reply:
Could not that be done just by setting proper error display in php?

Reply:
Where is it trying to open the connection? If something is throwing an error out of the box, this should be checked at the call site before trying to open the connection, not in the CouchDB wrapper.

Reply:
Inside the [dataquery] module, php/dataquery.class.inc has a setup() function where on ( line 54 ) the queryView automatically runs the open() function of the SocketWrapper. 

The goal was to hide the extra console warning being displayed because the LorisException prints a message to the page about how the database hasn't been configured. 

I'm not certain if a check to see if the database isn't using the default installation values in the setup() of dataquery.class.inc is the best approach. How should I proceed with it?